Framework/PythonInterface/mantid/api/src/Exports/SpectrumInfo.cpp

```diff
@@ -13,13 +18,53 @@ void export_SpectrumInfo() {
       .def("isMonitor", &SpectrumInfo::isMonitor, (arg("self"), arg("index")),
            "Returns true if the detector(s) associated with the spectrum are "
            "monitors.")
+
       .def("isMasked", &SpectrumInfo::isMasked, (arg("self"), arg("index")),
            "Returns true if the detector(s) associated with the spectrum are "
            "masked.")
+
+      .def("setMasked", &SpectrumInfo::setMasked,
+           (arg("self"), arg("index"), arg("masked")),
+           "Set the mask flag of the spectrum with the given index.")
+
       .def("hasDetectors", &SpectrumInfo::hasDetectors, (arg("self")),
            "Returns true if the spectrum is associated with detectors in the "
            "instrument.")
+
       .def("twoTheta", &SpectrumInfo::twoTheta, (arg("self"), arg("index")),
            "Returns the scattering angle 2 theta in radians w.r.t. beam "
-           "direction.");
+           "direction.")
+
+      .def("signedTwoTheta", &SpectrumInfo::signedTwoTheta,
+           (arg("self"), arg("index")),
+           "Returns the signed scattering angle 2 theta in radians w.r.t. beam "
+           "direction.")
+
+      .def("size", &SpectrumInfo::size, arg("self"),
+           "Returns the number of spectra.")
+
+      .def("hasUniqueDetector", &SpectrumInfo::hasUniqueDetector,
+           (arg("self"), arg("index")),
+           "Returns true if the spectrum is associated with exactly one "
+           "detector.")
+
+      .def("l1", &SpectrumInfo::l1, arg("self"),
+           "Returns the distance from the source to the sample.")
+
+      .def("l2", &SpectrumInfo::l2, (arg("self"), arg("index")),
+           "Returns the distance from the sample to the spectrum.")
+
+      .def("position", &SpectrumInfo::position, (arg("self"), arg("index")),
+           "Returns the position of the spectrum with the given index.")
```

Comment:
absolute position

Reply:
in fact anything mentioning `position` or `rotation` should be prefixed with absolute, to distinguish from `instrument 1.0`

Reply:
I see - I'll add in those changes